osi_referenceline.proto

```diff
@@ -0,0 +1,190 @@
+    // A.Y-B.Y)</code>, if A and B are the two points) between the point to
+    // describe and the nearest point on the polyline (this point might either
+    // be on a line segment or at an edge between two line segments). The
+    // distance is positive if the point is left of the polyline (in definition
+    // direction), negative if it is right of it.
+    // The S position of such a point outside the reference line is the same as
+    // the S value of the nearest point on the polyline.
+    //
+    // Notes:
+    // - The "nearest point on the polyline" is determined in 3D (even if the
+    //   resulting T value is only the 2D distance), in order to choose the
+    //   correct point for 3D curves (think reference lines for roads in parking
+    //   decks).
+    // - If there are several "nearest points", the one with the smallest S
+    //   coordinate on the polyline is chosen.
+    //
+    // ## Defining angles
+    //
+    // Sometimes an angle to a reference line is needed. This shall be defined
+    // as follows:
+    // First the nearest point on the polyline is determined, as described
+    // above. If this point is on a line segment, then the angle is calculated
+    // relative to the line segment on which the reference point lays.
+    // If the nearest point is at the edge between line segments, then the
+    // angle of the following line shall be chosen.
+    //
+    // ## Converting between world coordinates and ST coordinates
+    //
+    // The above rules define an ST coordinate system across the whole XY plane.
+    // Every XY position has a ST coordinate, but not necessarily a unique ST
+    // coordinate.
+    //
+    // Example:
+    // \image html OSI_ReferenceLine1.svg
+    //
+    // This shows a reference line (consisting of three points), and five points
+    // not on the reference line.
+    //
+    // - For \c P1, the situation is clear, since there is exactly one nearest
+    //   point on the polyline. The resulting ST coordinate uniquely maps back
+    //   to \c P1.
+    // - \c P2 has multiple points "nearest points" on the polyline.
+    //   As can be seen here, two  ST coordinates map to \c P2 (red and grey
+    //   dotted line).  Following the rules above, the one with the smallest S
+    //   value is chosen (the red dotted line).
+    // - \c P3 has a unique "nearest point" on the polyline. However, multiple
+    //   points map to the same ST coordinate as that of \c P3, e.g. \c P4
+    //   (drawn in grey).
+    // - Finally, \c P5 shows how the reference line is extended infinitely for
+    //   points that are "outside" the reference line.
+    //
+    // The sampling of the polyline must be chosen such that the error
+    // when converting coordinates is "small enough". The exact needed
+    // precision is defined for each user, where the reference line is
+    // referenced.
+    //
+    // ## Various notes
+    //
+    // Notes on OpenDRIVE compatibility:
```

Comment:
The only thing I am missing here is any advise regarding how the Reference Line should be handled when the source is now OpenDRIVE. Is the intention or recommendation that similar rules to OpenDRIVE are followed? e.g. on a bi-directional road, the reference line is between the lanes of opposing direction and on uni-directional roads, the reference line is nearest to the central division between the opposing direction.

If this is the case, I would find it beneficial to state that here, as the Reference Line is an integral part of the Logical Lane structure.

Reply:
@ndunningbmw Good question. I added a note at https://github.com/OpenSimulationInterface/open-simulation-interface/pull/599/commits/c3c0244f392938d4cbee073eeecb539d5261de6d . Does that answer your question?

Reply:
Looks good to me!

Reply:
Thanks for the fast response :) Closing this then.